content/blog/2020-03-02-react-v16.13.0.md

```diff
@@ -0,0 +1,154 @@
+
+The fix is usually to move this state change into a `useEffect(...)`.
+
+### Warnings for some deprecated string refs
+
+[String Refs is an old legacy API](https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs) which is discouraged and is going to be deprecated in the future. This release adds a new warning to some unusual edge cases that can't be fixed automatically. For example using a string ref in a render prop:
+
+```jsx
+class ClassWithRenderProp extends React.Component {
+  componentDidMount() {
+    something(this.refs.foo);
+  }
+  render() {
+    return this.props.children();
+  }
+}
+
+class ClassParent extends React.Component {
+  render() {
+    return (
+      <ClassWithRenderProp>{() => <View ref="foo" />}</ClassWithRenderProp>
+    );
+  }
+}
+```
+
+You most likely don't have code like this but if you do, we recommend that you convert it to [`React.createRef()`](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) instead:
+
+```jsx
+class ClassWithRenderProp extends React.Component {
+  foo = React.createRef();
+  componentDidMount() {
+    something(this.foo.current);
+  }
+  render() {
+    return this.props.children(this.foo);
+  }
+}
+
+class ClassParent extends React.Component {
+  render() {
+    return (
+      <ClassWithRenderProp>{foo => <View ref={foo} />}</ClassWithRenderProp>
+    );
+  }
+}
+```
+
+Fixing these edge cases will be a prerequisite to do before being able to apply an automatic codemod.
+
+> To get this warning you need to have the [babel-plugin-transform-react-jsx-self](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx-self) installed in your Babel plugins. 
+> 
+> If you use the “react” preset you should already have this installed by default!
+
+### Renaming `ReactDOM.unstable_createPortal` to `ReactDOM.createPortal`
+
+Portals are now considered first class supported APIs, and as such we're removing the `unstable_` prefix for the method that creates portals. Replace all calls to `ReactDOM.unstable_createPortal` with `ReactDOM.createPortal`.
+
+### Deprecating `React.createFactory`
+
+[`React.createFactory`](https://reactjs.org/docs/react-api.html#createfactory) is a legacy helper for creating React elements. This release adds a deprecation warning to the method, and will be removed in a future major version. Replace usages of `React.createFactory` with regular JSX.
+
+### Other warnings
+
+- When dynamically changing `style` on a dom element, it's sometimes possible that the applied style clashes with a previously set style, resulting in a badly styled element. React now detects this pattern and logs a warning. The workaround is to pass the same object 'shape' to `style` everytime. 
```

Comment:
I don't think we need quotes around shape here. Either we're sure the reader will understand us, or we need to be more specific.

Reply:
rewrote this section